src/test/java/org/reactome/addlinks/test/TestHPAFileProcessor.java

```diff
@@ -0,0 +1,54 @@
+package org.reactome.addlinks.test;
+
+import org.junit.Test;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.reactome.addlinks.fileprocessors.HPAFileProcessor;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.fail;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Map;
+
+@PrepareForTest({com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl.class})
+
+public class TestHPAFileProcessor {
+
+    private static final String TEST_DIRECTORY = "src/test/resources/";
+    private static final String TEST_FILE = "hpa-test.tsv";
+    private static final int EXPECTED_TEST_HPA_MAPPING_SIZE = 9;
+
+    @Test
+    public void testHPAFileProcessor() throws IOException {
+
+        HPAFileProcessor processor = new HPAFileProcessor("test");
+        processor.setPath(Paths.get(TEST_DIRECTORY, TEST_FILE + ".zip"));
+        Map<String, List<String>> testMapping = processor.getIdMappingsFromFile();
+
+        assertThat(testMapping.size(), is(equalTo(EXPECTED_TEST_HPA_MAPPING_SIZE)));
+        assertThat(testMapping.get("P08603").get(0), is(equalTo("ENSG00000000971-CFH")));
+
+        Files.delete(Paths.get(TEST_DIRECTORY, TEST_FILE, TEST_FILE));
+        Files.delete(Paths.get(TEST_DIRECTORY, TEST_FILE));
```

Comment:
Is the path in the first line a typo?

Reply:
No, for zipped files it actually creates a directory from the archive. If you look on the release server you should see examples of this for `proteinatlas.tsv.zip` (the HPA file) and 'hmdb_metabolites/proteins.zip`. The structure is proteinatlas.tsv/proteinatlas.tsv. I agree it looks a little odd but I'm not sure if there is a way around this? (Paging @SolomonShorser-OICR)

Reply:
I really don't recall the specific of what HMDB files do off the top of my head, but just by looking at the code, I suspect that the zip file contains a directory named `${TEST_FILE}`, and that directory contains a file named `${TEST_FILE}`. So yeah it looks weird, but if it actually works, then it's probably not a typo. I would guess that deleting the the directory named `${TEST_FILE}` without deleting the contents first will result in some sort of file system/operating system error.

Reply:
Okay, sounds good.  I just wanted to check.